### PR TITLE
add parsing of centos pull-request comments and updates

### PR DESCRIPTION
**Pull request by sakalosj**
Added parsing of centos pull-request comments and updates.

### CODE REVIEW

File: packit_service/worker/jobs.py [hunk 1]
```diff
@@ -271,6 +279,13 @@ def process_comment_jobs(
                 success=True, details={"msg": "Account is not whitelisted!"}
             )
 
+        # VERY UGLY
+        # TODO: REFACTOR !!!
+        if handler_kls == GitHubPullRequestCommentCoprBuildHandler and isinstance(
+            event, PullRequestCommentPagureEvent
+        ):
+            handler_kls = PagurePullRequestCommentCoprBuildHandler
+
```
Comment (sakalosj) [inline packit_service/worker/jobs.py:288]: this is very ugly, but enough for my testing and probably can stay here till refactoring.
any suggestions are welcome

Comment (lachmanfrantisek) [reply packit_service/worker/jobs.py:288]: Some nicer way would require lists in `MAP_COMMENT_ACTION_TO_HANDLER` and making some logic on top of it.

I would use this approach for now and discuss after summit -- I'll add a note to our refactor agenda about this.

Comment (lachmanfrantisek) [reply packit_service/worker/jobs.py:288]: I think we already discuss that, but do we want to use whitelist for centos-stream? Can we move this above the whitelisting code?

File: packit_service/worker/handlers/centosmsg_handlers.py [hunk 2]
```diff
@@ -142,3 +147,35 @@ def pre_check(self) -> bool:
...
+    event: PullRequestCommentPagureEvent
+
+    def __init__(self, config: ServiceConfig, event: PullRequestCommentPagureEvent):
+        super().__init__(config=config, event=event)
+        self.config = config
+        self.event = event
+        self.project: GitProject = event.get_project()
+        # Get the latest pull request commit
+        self.event.commit_sha = self.project.get_pr(self.event.pr_id).head_commit
+        self.event.base_ref = self.event.commit_sha
```
[the remark this answers is not in the document]
Comment (sakalosj) [reply packit_service/worker/handlers/centosmsg_handlers.py:168]: followed [github implementation](https://github.com/packit-service/packit-service/blob/d86f25073029cd243767aeeb3e85d22f60c03ddb/packit_service/worker/handlers/github_handlers.py#L572-L592) counterpart here. will fix it

File: packit_service/worker/jobs.py (hunk 1 above)
Comment (sakalosj) [reply packit_service/worker/jobs.py:288]: yes u are right, will fix it

File: packit_service/worker/handlers/centosmsg_handlers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (lachmanfrantisek) [reply packit_service/worker/handlers/centosmsg_handlers.py:168]: Thanks!

**Comment by TomasTomecek** [comment]
uhhh, is github having issues again? I cannot reply to the inline comments

> I think we already discuss that, but do we want to use whitelist for centos-stream? Can we move this above the whitelisting code?

We don't need whitelist for the Stream.

**Comment by lachmanfrantisek** [comment]
> uhhh, is github having issues again? I cannot reply to the inline comments

Neither can I...;(

File: packit_service/worker/handlers/centosmsg_handlers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (sakalosj) [reply packit_service/worker/handlers/centosmsg_handlers.py:168]: fixed in parser

File: packit_service/worker/jobs.py (hunk 1 above)
Comment (sakalosj) [reply packit_service/worker/jobs.py:288]: fixed  - moved logic to Whitelist to avoid whitelisting

File: packit_service/worker/whitelist.py [hunk 3]
```diff
@@ -190,8 +191,17 @@ def check_and_report(
         :param project: GitProject
         :return:
         """
+
+        # whitelist checks dont apply to CentOS (Pagure)
+        if isinstance(
+            event,
+            (PushPagureEvent, PullRequestPagureEvent, PullRequestCommentPagureEvent),
+        ):
+            logger.info("Centos (Pagure) events dont require whitelist checks")
```
Comment (lachmanfrantisek) [inline packit_service/worker/whitelist.py:200]: We need to think about this more when working on the Pagure support as upstream (#556).

Comment (sakalosj) [reply packit_service/worker/whitelist.py:200]: yes, i am aware of that, and in future we can achieve CentOS stream events exclusion using some Event attribute 

e.g. our centos listener is adding topic field to message dictionary which is sent for processing. it consists of sourceurl+gitevent -> sourceurl can be used for this purpose 

i think that at this point, there is no reason to implement it.

Comment (sakalosj) [reply packit_service/worker/whitelist.py:200]: should i implement it in this PR?

File: packit_service/worker/handlers/centosmsg_handlers.py [hunk 4]
```diff
@@ -142,3 +147,33 @@ def pre_check(self) -> bool:
...
+    type = CommentAction.copr_build
+    triggers = [TheJobTriggerType.pr_comment]
+    event: PullRequestCommentPagureEvent
+
+    def __init__(self, config: ServiceConfig, event: PullRequestCommentPagureEvent):
+        super().__init__(config=config, event=event)
+        self.config = config
+        self.event = event
+        self.project: GitProject = event.get_project()
+        # Get the latest pull request commit
```
[the remark this answers is not in the document]
Comment (sakalosj) [reply packit_service/worker/handlers/centosmsg_handlers.py]: fixed

File: packit_service/worker/whitelist.py (hunk 3 above)
Comment (lachmanfrantisek) [reply packit_service/worker/whitelist.py:200]: Not need to solve that here.

**Pull request merged**